crates/core/src/sql/statements/define/namespace.rs

```diff
@@ -48,22 +48,14 @@ impl DefineNamespaceStatement {
 		}
 		// Process the statement
 		let key = crate::key::root::ns::new(&self.name);
-		txn.set(
-			key,
-			revision::to_vec(&DefineNamespaceStatement {
-				id: if self.id.is_none() {
-					Some(txn.lock().await.get_next_ns_id().await?)
-				} else {
-					None
-				},
```

Comment:
This appears to have been a bug. When the Namespace ID is set we basically throw it away. This means that if we try to migrate to having a Catalog and looking up Namespaces (and Databases and Tables) by their IDs instead of their names we may not be able to look up the ID from the KV Store because it'll just be `None`.

@tobiemh or @kearfy if I want to do a migration to querying based off ID, how should an entry which is missing the ID be handled? Should we just call `get_next_ns_id()` and populate it with that?

Reply:
I addressed this in another #5819.